### ISSUE TITLE
pyjoulescope_examples/bin/trigger.py - 'dict' object has no attribute 'astype'

### ISSUE THREAD

**Issue by anujfalcon** [jetperch/pyjoulescope_examples#8]
Given:stream_buffer.samples_get(start_id, end_id, fields=field)
Working Change: stream_buffer.samples_get(start_id, end_id, fields=[field])

Given: gpi.astype(np.int8)
Working Change: gpi = gpi['signals'][field]['value'].astype(np.int8)

Tested on:

Python: 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]
    Platform: Linux-5.15.0-67-generic-x86_64-with-glibc2.35 (linux)
    Processor: x86_64
joulescope version: 1.1.2

**Comment by mliberty1** [comment]
Hi @anujfalcon and thank you for reporting this issue!  The actual problem is that the newer v1 pyjoulescope implementation does not correctly support single field string names.  According to the samples_get API:

```
        :return: The dict containing top-level 'time' and 'signals' keys.
            The 'time' value is a dict contain the timing metadata for
            these samples.  The 'signals' value is a dict with one
            key for each field in fields.  Each field value is also
            a dict with entries 'value' and 'units'.
            However, if single field string is provided to fields, then just
            return that field's value.
```

Here is the fix: https://github.com/jetperch/pyjoulescope/commit/f2e257f8a515297199d30080dee99d61a53d1bf1

We will publish an official release to pypi later this week.

**Comment by mliberty1** [comment]
In the meantime, your workaround looks good!

**Comment by anujfalcon** [comment]
The fix throws "UnboundLocalError: local variable 'is_single_result' referenced before assignment" when the input to fields parameter in stream_buffer.samples_get(start_id, end_id, fields=[field]) is a list, which shouldn't be the case right?
Working addition: is_single_result = False @ line 357 in joulescope/v1/stream_buffer.py

**Comment by mliberty1** [comment]
Doh!  Fixed in [42c34f0](https://github.com/jetperch/pyjoulescope/commit/42c34f0ee6ec3ccfa1cb219d6ddc4160166600bb). This will be part of joulescope 1.1.3 when it is released this week.

I am going to close this issue, but please reopen if you continue to see an issue.
